web/src/beta/lib/reearth-ui/components/PopupMenu/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { FC, useCallback, useState } from "react";
+import { Link } from "react-router-dom";
+
+import { Popup, Icon, Typography, IconName } from "@reearth/beta/lib/reearth-ui";
+import { styled, useTheme } from "@reearth/services/theme";
+
+const MULTLEVEL_OFFSET = 6;
+const DEFAULT_OFFSET = 4;
+const DEFAULT_MENU_WIDTH = 180;
+
+export type Items = {
+  title?: string;
+  path?: string;
+  icon?: IconName;
+  subItem?: Items[];
+  onClick?: () => void;
+};
+
+export type PopupMenuProps = {
+  label?: string;
+  icon?: IconName;
+  menu: Items[];
+  nested?: boolean;
+  width?: number;
+};
+
+export const PopupMenu: FC<PopupMenuProps> = ({ label, menu, nested, width, icon }) => {
+  const [open, setOpen] = useState(false);
+  const [hoveredItemIndex, setHoveredItemIndex] = useState<number | null>(null);
+  const theme = useTheme();
+
+  const handlePopOver = useCallback(() => setOpen(!open), [open]);
+
+  const renderMenuItems = (menuItems: Items[]) => {
+    return (
+      <PopupMenuWrapper width={width}>
+        {menuItems.map(({ title, path, icon, subItem, onClick }, index) => (
+          <Item
+            key={index}
+            onMouseEnter={() => setHoveredItemIndex(index)}
+            onMouseLeave={() => setHoveredItemIndex(null)}
+            isHovered={hoveredItemIndex === index}
+            onClick={onClick}>
```

Comment:
Here onClick needs to have a prop like `id`.
```suggestion
            onClick={() => onClick(id)}>
```